apps/weather_status/lib/Service/WeatherStatusService.php

```diff
@@ -0,0 +1,283 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2020, Julien Veyssier
+ *
+ * @author Julien Veyssier <eneiluj@posteo.net>
+ *
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program. If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\WeatherStatus\Service;
+
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\IConfig;
+use OCP\IL10N;
+use OCP\App\IAppManager;
+use OCP\Accounts\IAccountManager;
+use OCP\Accounts\PropertyDoesNotExistException;
+use OCP\IUserManager;
+use OCP\Http\Client\IClientService;
+use OCP\ICacheFactory;
+
+/**
+ * Class StatusService
+ *
+ * @package OCA\WeatherStatus\Service
+ */
+class WeatherStatusService {
+	public const MODE_BROWSER_LOCATION = 1;
+	public const MODE_MANUAL_LOCATION = 2;
+
+	/**
+	 * StatusService constructor.
+	 *
+	 * @param ITimeFactory $timeFactory
+	 * @param PredefinedStatusService $defaultStatusService,
+	 */
+	public function __construct(ITimeFactory $timeFactory,
+								IClientService $clientService,
+								IConfig $config,
+								IL10N $l10n,
+								IAccountManager $accountManager,
+								IUserManager $userManager,
+								IAppManager $appManager,
+								ICacheFactory $cacheFactory,
+								string $userId) {
+		$this->timeFactory = $timeFactory;
+		$this->config = $config;
+		$this->userId = $userId;
+		$this->l10n = $l10n;
+		$this->accountManager = $accountManager;
+		$this->userManager = $userManager;
+		$this->appManager = $appManager;
+		$this->version = $appManager->getAppVersion('weather_status');
```

Comment:
```suggestion
		$this->version = $appManager->getAppVersion(Application::APP_ID);
```

Same goes for the other occurences?
Just cleanup :)